package-res/reportviewer/reportviewer.js

```diff
@@ -863,30 +875,44 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
+                      var urlContent2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/content';
+                      logger && logger.log("Will set iframe url to " + urlContent2.substr(0, 50) + "... ");
+
+                      $('#hiddenReportContentForm').attr("action", urlContent2);
+                      $('#hiddenReportContentForm').submit();
+                      $('#reportContent').attr("data-src", urlContent2);
+                      this._updatedIFrameSrc = true;
+                      dlg.hide();
+                      isIframeContentSet = true;
+                    }
+                  });
+
                   if(isFirstContAvStatus) {
                     isFirstContAvStatus = false;
-                    var handleContAvailCallback = dojo.hitch(this, function(result2) {
-                      resultJson2 = JSON.parse(result2);
-                      if(resultJson2.status == "QUEUED" || resultJson2.status == "WORKING") {
-                        var urlStatus2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/status';
-                        setTimeout(function(){ pentahoGet(urlStatus2, "", handleContAvailCallback); }, this.reportPrompt._pollingInterval);
-                      } else if (resultJson2.status == "FINISHED") {
-                        var urlContent2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/content';
-                        logger && logger.log("Will set iframe url to " + urlContent2.substr(0, 50) + "... ");
- 
-                        $('#hiddenReportContentForm').attr("action", urlContent2);
-                        $('#hiddenReportContentForm').submit();
-                        $('#reportContent').attr("data-src", urlContent2);
-                        this._updatedIFrameSrc = true;
-                        dlg.hide();
-                        isIframeContentSet = true;
-                      }
-                    });
-                    //get first page
-                    pentahoGet('reportjob', url.substring(url.lastIndexOf("/report?")+"/report?".length, url.length), handleContAvailCallback, 'text/text');
+
+                    if(this._currentStoredPagesCount > me.view._getAcceptedPage()){
+                      pentahoGet('reportjob', url.substring(url.lastIndexOf("/report?")+"/report?".length, url.length), handleContAvailCallback, 'text/text');
+                    }
+
                     var urlStatus = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson.uuid + '/status';
                     setTimeout(function(){ pentahoGet(urlStatus, "", handleResultCallback); }, this.reportPrompt._pollingInterval);
                   } else {
+
+                    if( (this._requestedPage > 0) && (this._currentStoredPagesCount > this._requestedPage) ){
```

Comment:
requestedPage has an initial value of "-1" to signal no selection. Then here you test for > 0, which means there is a possible value "0" that is not covered. Assuming the page-control cannot have 0 as valid value, why not use that instead of -1.